### PR TITLE
Skip yaml content for skipped testcases

### PR DESCRIPTION
**Pull request by Siebe-Nijs**
## Summary by CodeRabbit

* **New Features**
  * Test reports now properly exclude extra content entries for skipped test cases, improving report clarity

* **Tests**
  * Added acceptance tests validating proper handling of extra content for both normal and skipped test cases
  * Added edge case testing to catch invalid XML structure errors

* **Chores**
  * Updated .gitignore to exclude Visual Studio Code editor configuration directory

[Review Change Stack](https://app.coderabbit.ai/change-stack/melexis/xunit2rst/pull/54)

### CODE REVIEW

File: mlx/xunit2rst/xunit2rst.py [hunk 1]
```diff
@@ -67,7 +96,9 @@ def generate_xunit_to_rst(input_file, rst_file, itemize_suites, failure_message,
         report_name = report_name[:-len('_report')]
 
     indexed_extra_content_map = {}
+    all_suites = list(test_suites)
     for i, file in report_info_files.items():
+        suite = all_suites[i]
         extra_content_map = {}
```
[the remark this answers is not in the document]
Comment (Siebe-Nijs) [reply mlx/xunit2rst/xunit2rst.py:102]: I do not see the benefit of this change. The xunit root function specifically returns a report info file list with the same indexes as the suites.

**Comment by JokeWaumans** [comment]
Coderabbitai is almost correct about https://github.com/melexis/xunit2rst/pull/54/changes#r3226542414 (For now there is no IndexError but a KeyError.

I added a testcase in ba844ef612fce6b4cfed80b5b7a887c78d70f498:
`test_edge_case_with_nontestsuite_element`
1. The XML file contains `[<testsuite name="Suite1">, <foo/>, <testsuite name="Suite2">]`.
2. The `for i, suite in enumerate(test_suites):` loop of `parse_xunit_root` starts.
3. `i=0`: It processes `<testsuite name="Suite1">`. It finds the xunit2rst content file property and correctly adds an entry for index 0 to `report_info_files`. The `<properties>` tag is then removed from this suite.
4. `i=1`: It processes `<foo/>`. Since `if suite.tag != 'testsuite':` condition is true, `test_suites.remove(suite)` is executed. The list of children is immediately modified during the iteration. It now contains only two elements: `[<testsuite name="Suite1">, <testsuite name="Suite2">]`.
5. The Bug: The for loop's internal counter/pointer is now in an inconsistent state. It has just processed the item at index 1. To continue, it moves to what it thinks is the next position, index 2. However, since the list was shortened, there is no longer an element at index 2. The loop sees it has reached the end of the (now shorter) list and terminates.

The `<testsuite name="Suite2">`, which was originally at index 2, is never even visited by the loop.
This is why `report_info_files` only contains `{0: ...}` and why the unprocessed `<testsuite name="Suite2">` (still containing its `<properties>` child) is passed to the template, causing the `KeyError` (there is no `name` attribute in `properties`).

When changing all for loop to use a "copy" for example change `for i, suite in enumerate(test_suites):` to `for i, suite in enumerate(list(test_suites)):` and do the same for the other loop inside that for loop, it will now generate the `IndexError: list index out of range` about line `suite = all_suites[i]` (where Coderabbit was talking about)

File: mlx/xunit2rst/xunit2rst.py [hunk 2]
```diff
@@ -67,7 +96,9 @@ def generate_xunit_to_rst(input_file, rst_file, itemize_suites, failure_message,
         report_name = report_name[:-len('_report')]
 
     indexed_extra_content_map = {}
+    all_suites = list(test_suites)
     for i, file in report_info_files.items():
```
Comment (JokeWaumans) [inline mlx/xunit2rst/xunit2rst.py:100]:
```suggestion
    for i, (suite, file) in report_info_files.items():
```

File: mlx/xunit2rst/xunit2rst.py
Comment (JokeWaumans) [inline mlx/xunit2rst/xunit2rst.py:188]:
```suggestion
    for i, suite in enumerate(list(test_suites)):
```

Comment (JokeWaumans) [inline mlx/xunit2rst/xunit2rst.py:189]:
```suggestion
        for element in reversed(list(suite)):
```

Comment (JokeWaumans) [inline mlx/xunit2rst/xunit2rst.py:195]:
```suggestion
        for element in list(suite):
```

Comment (JokeWaumans) [inline mlx/xunit2rst/xunit2rst.py:199]:
```suggestion
                    rreport_info_files[i] = (suite, value)
```

File: tests/acceptance_test.py [hunk 3]
```diff
@@ -330,6 +350,14 @@ def test_suite_header_content_with_mako_template(self):
 
         assert_output_matches_reference(output_rst, rst_file_name)
 
+    def test_edge_case_with_nontestsuite_element(self):
+        file_name = 'edge_case'
+        xml_file_name = '{}.xml'.format(file_name)
+        input_xml = str(TEST_IN_DIR / xml_file_name)
+        output_rst = str(TEST_OUT_DIR / 'edge_case.rst')
+        with self.assertRaises(KeyError):
+            xunit2rst_check(input_xml, output_rst)
```
Comment (JokeWaumans) [inline tests/acceptance_test.py]:
```suggestion
        xunit2rst_check(input_xml, output_rst)
```

**Review by JokeWaumans — COMMENTED** [review]
prefix_test.py should also change to fix tests:

`self.assertEqual(content_files, {3: Path("../../doc/source/extra_content.yml.mako")})`
to
```
self.assertIn(3, content_files)
suite, path = content_files[3]
self.assertEqual(suite.tag, 'testsuite')
self.assertEqual(path, Path("../../doc/source/extra_content.yml.mako"))
```

and

`self.assertEqual(content_files, {0: Path('./extra_content1.yml')})`
to
```
self.assertIn(0, content_files)
suite, path = content_files[0]
self.assertEqual(suite.tag, 'testsuite')
self.assertEqual(path, Path('./extra_content1.yml'))
```

**Comment by Siebe-Nijs** [comment]
I'm going to revert most of these, and add a checker in parse_xunit_root to remove non test suites in there.

File: mlx/xunit2rst/xunit2rst.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Siebe-Nijs) [reply mlx/xunit2rst/xunit2rst.py:102]: With the latest changes is this issue fixed now?

**Comment by Siebe-Nijs** [comment]
Should be ready now

**Review by JokeWaumans — APPROVED** [review]
